hatch_pip_compile/plugin.py

```diff
@@ -92,17 +101,20 @@ def _hatch_pip_compile_install(self):
             )
```

Comment:
pip-tools is needed for:
* lock - only if lock file out of date
* install - only for pip-sync installer

as such, installing pip-tools is not always necessary now :thinking: